src/matrix_ordering.jl

```diff
@@ -0,0 +1,49 @@
+function pseudo_peripheral_node(g::SimpleGraph, src)
+    T = eltype(g)
+    u::T = src
+    lp = 0
+    while true
```

Comment:
Isn't there a problem here, when the graph contains disconnected components? I think in this case, `lp` will be set to `typemax(T)` in the first iteration and therefore the loop will stop in the second iteration.

Reply:
yes i should change it to check only the reachable vertices (or make gdistance 0 where it returns typemax)